v0/destinations/google_adwords_remarketing_lists/transform.js

```diff
@@ -22,7 +22,7 @@ const {
 
 const hashEncrypt = object => {
   Object.keys(object).forEach(key => {
-    if (hashAttributes.includes(key)) {
+    if (hashAttributes.includes(key) && object[key]) {
```

Comment:
if `object[key]` is undefined/null we are skipping here. Hence only hashing is skipped but the `null` / `undefined` values are sent to google. Does google accept those?
Secondly, if an identifier is having null/undefined shouldn't we skip sending those to google entirely?

Reply:
No we are not sending null values we are discarding those values.

Reply:
It's also skipping all the "falsy" values a.k.a `undefined`, `NaN`, ``, `false`, and `0`.

Reply:
What are the repercussions if we don't send the invalid values to the destination?

Reply:
There are not any repercussions if we don't send the invalid values.

Reply:
I think the targeting will be weak as it will be hard to identify a person with less identifiable attributes

Reply:
There is nothing to do with us as the customer is unable to collect the user data properly.

Reply:
Re-iterating this,

> It's also skipping all the "falsy" values a.k.a undefined, NaN, ``, false, and `0`.

Reply:
The values we are sending as user traits, can't be any falsy value.

Reply:
Why can't a trait be `false` or `0`?

Reply:
somebody can send `email` : 0, in that case we will skip it As Far As I understand

Reply:
Makes sense but we can't apply that to all the traits.
Ex: 
`numberOfVisits`: `0`
`isActive`: `false`

Reply:
Yes agreed, but from lines 78 to 117 as we are filtering over user identifiers we are safe there.

Reply:
@ItsSudip can you confirm

Reply:
@saikumarrs we are not sending this type of trait. Google Adwords required only the following [userIdentifiers](https://developers.google.com/google-ads/api/reference/rpc/v10/UserIdentifier).

Reply:
Sounds good.